micro-rdk/src/common/config.rs

```diff
@@ -20,6 +23,72 @@ pub enum AttributeError {
     ValidationError(String),
 }
 
+impl TryFrom<&DeviceAgentConfigResponse> for AgentConfig {
+    type Error = AttributeError;
+    fn try_from(value: &DeviceAgentConfigResponse) -> Result<Self, Self::Error> {
+        if let Some(ref additional_networks) = value.additional_networks {
+            let network_settings = additional_networks
+                .fields
+                .iter()
+                .filter_map(|(_k, v)| {
+                    let local_kind: Option<Kind> =
+                        v.kind.clone().and_then(|v| Kind::try_from(v).ok());
+                    local_kind
+                        .as_ref()
+                        .and_then(|v| NetworkSetting::try_from(v).ok())
+                })
+                .collect::<Vec<NetworkSetting>>();
+            Ok(Self { network_settings })
+        } else {
+            Err(AttributeError::ConversionImpossibleError)
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct AgentConfig {
+    pub network_settings: Vec<NetworkSetting>,
+}
+
+pub struct NetworkSetting {
+    ssid: String,
+    password: String,
+    priority: usize,
```

Comment:
I believe priority is permitted to be negative, so I don't think a `usize` is the right type.